File: server/app/models/agent.py
from sqlalchemy import Column, Integer, String, Text, DateTime
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database import Base


class Agent(Base):
    __tablename__ = "agents"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(255), nullable=False)
    system_prompt = Column(Text, nullable=False, default='')
    llm_config_id = Column(Integer, nullable=False, index=True)
    description = Column(Text, nullable=False, default='')
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

    llm_config = relationship(
        "LLMConfig",
        back_populates="agents",
        primaryjoin="Agent.llm_config_id == LLMConfig.id",
        foreign_keys="Agent.llm_config_id"
    )

File: server_go/internal/model/agent.go
package model

import "time"

type Agent struct {
	ID               int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	Name             string    `gorm:"type:varchar(255);not null" json:"name"`
	CharacterSettings string   `gorm:"type:text;not null;default:'';column:character_settings" json:"character_settings"`
	LLMConfigID      int64     `gorm:"not null;index;column:llm_config_id" json:"llm_config_id"`
	EmbeddingConfigID int64    `gorm:"not null;default:0;index;column:embedding_config_id" json:"embedding_config_id"`
	Description      string    `gorm:"type:text;not null;default:''" json:"description"`
	Avatar           string    `gorm:"type:varchar(500);not null;default:''" json:"avatar"`
	CreatedAt        time.Time `gorm:"not null;autoCreateTime" json:"created_at"`
	UpdatedAt        time.Time `gorm:"not null;autoUpdateTime" json:"updated_at"`
}

func (Agent) TableName() string { return "agents" }
